internal/install/packs/packs_installer.go

```diff
@@ -0,0 +1,137 @@
+package packs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+
+	"github.com/newrelic/newrelic-cli/internal/credentials"
+	"github.com/newrelic/newrelic-cli/internal/install/types"
+	"github.com/newrelic/newrelic-cli/internal/install/ux"
+	"github.com/newrelic/newrelic-client-go/newrelic"
+	"github.com/newrelic/newrelic-client-go/pkg/dashboards"
+	"github.com/newrelic/newrelic-client-go/pkg/entities"
+	log "github.com/sirupsen/logrus"
+)
+
+// Maybe both fetchPacks and installPacks should be one interface
+// vs separate interfaces?
+// nolint: golint
+type PacksInstaller interface {
+	Install(ctx context.Context, packs []types.OpenInstallationObservabilityPack) error
+}
+
+type ConcretePacksInstaller struct {
+	client            *newrelic.NewRelic
+	progressIndicator ux.ProgressIndicator
+}
+
+func NewConcretePacksInstaller(client *newrelic.NewRelic) *ConcretePacksInstaller {
+	return &ConcretePacksInstaller{
+		client:            client,
+		progressIndicator: ux.NewPlainProgress(),
+	}
+}
+
+func defaultHTTPGetFunc(dashboardURL string) (*http.Response, error) {
+	return http.Get(dashboardURL)
+}
+
+func (p *ConcretePacksInstaller) Install(ctx context.Context, packs []types.OpenInstallationObservabilityPack) error {
+	msg := "Installing observability packs"
+	p.progressIndicator.Start(msg)
+	defer p.progressIndicator.Stop()
+
+	for _, pack := range packs {
+		fmt.Printf("\n  Installing pack: %s\n", pack.Name)
+
+		// Only installing dashboards currently
+		if pack.Dashboards != nil {
+			for _, dashboard := range pack.Dashboards {
+				if _, err := p.createObservabilityPackDashboard(ctx, dashboard); err != nil {
+					p.progressIndicator.Fail(msg)
+					return fmt.Errorf("Failed to create observability pack dashboard [%s]: %s", dashboard.Name, err) // nolint: golint
+				}
+			}
+		}
+	}
+
+	p.progressIndicator.Success(msg)
+	return nil
+}
+
+func (p *ConcretePacksInstaller) createObservabilityPackDashboard(ctx context.Context, d types.OpenInstallationObservabilityPackDashboard) (*dashboards.DashboardCreateResult, error) {
+	defaultProfile := credentials.DefaultProfile()
+	accountID := defaultProfile.AccountID
+
+	body, err := getJSONfromURL(d.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	dashboard, err := transformDashboardJSON(body, defaultProfile.AccountID)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("  ==> Creating dashboard: %s\n", dashboard.Name)
+
+	// Check for existence of dashboard before creating a new one
+	dashboards, err := p.client.Dashboards.ListDashboards(&dashboards.ListDashboardsParams{
```

Comment:
This will break when the Insights EOL happpens.  There is no Nerdgraph-based replacement in the client yet.  I'll create an issue for that work.